### PR TITLE
Clarify expectations for overlay creation

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** meta documentation

This is a clarification PR. It does not change the policy, it just highlights that developers / contributors should not expect to be the ones to do all the work when overlays are needed for breaking changes. The need for clarification was made quite clear by comments by some core developers at #11494.

There are other small improvements to the wording that sneaked in this PR.

### CODE REVIEW

File: dev/ci/README-developers.md [hunk 1]
```diff
@@ -43,9 +43,10 @@ See also [`test-suite/README.md`](../../test-suite/README.md) for information ab
 ### Breaking changes
 
 When your PR breaks an external project we test in our CI, you must
-prepare a patch (or ask someone to prepare a patch) to fix the
-project. There is experimental support for an improved workflow, see
-[the next section](#experimental-automatic-overlay-creation-and-building), below
+prepare a patch (or ask someone —possibly the project author— to
```
Comment (JasonGross) [inline dev/ci/README-developers.md]:
```suggestion
prepare a patch (or ask someone—possibly the project author—to
```
I think generally there are either spaces on both sides, or spaces on neither side; I've never seen dashes like this with spaces only on one side (I've only seen that when combining the suffix of two hypenated words, such as in "brother- and sister-in-law")

File: dev/ci/README-users.md [hunk 2]
```diff
@@ -75,21 +79,21 @@ Have a look at [#7656](https://github.com/coq/coq/pull/7656/files) for an
...
+your project for particular Coq versions. This is in fact very likely
+if your project includes a Coq ML plugin.
 
-We thus recommend a branching convention that mirrors Coq's branching
-policy. Then, you would have a `master` branch that follows Coq's
-`master`, a `v8.8` branch that works with Coq's `v8.8` branch and so
-on.
+For such projects, we recommend a branching convention that mirrors
+Coq's branching policy. Then, you would have a `master` branch that
+follows Coq's `master`, a `v8.8` branch that works with Coq's `v8.8`
```
Comment (JasonGross) [inline dev/ci/README-users.md]: I thought this was no longer the recommended policy, and the policy was to instead use `coq-v8.8` or `coq-8.8` or `$project_version-coq-8.8` or `$project_version-coq-v8.8`?

File: dev/ci/README-developers.md (hunk 1 above)
Comment (Zimmi48) [reply dev/ci/README-developers.md]: Note that these are em-dashes, so not the same character as in "brother-in-law". You might still be right though.

File: dev/ci/README-users.md (hunk 2 above)
Comment (Zimmi48) [reply dev/ci/README-users.md]: I think that this was introduced by @ejgallego and that the exact recommendation was debated at the time and not really settled. I believe that the use of the exact same branch name is still what @ejgallego's tools expect. My preference would indeed be to recommend using the `coq-` prefix and have tools support all these standards. Anyway, that is kind of unrelated to the current PR, which is not modifying the content here.

File: dev/ci/README-developers.md (hunk 1 above)
Comment (JasonGross) [reply dev/ci/README-developers.md]: Yes, not the same character.  But see, e.g., https://www.thepunctuationguide.com/em-dash.html
> Spaces with the em dash
> The em dash is typically used without spaces on either side, and that is the style used in this guide. Most newspapers, however, set the em dash off with a single space on each side. Example Most newspapers — and all that follow AP style — insert a space before and after the em dash.

File: dev/ci/README-users.md (hunk 2 above)
Comment (ejgallego) [reply dev/ci/README-users.md]: At least for ML plugins, I still prefer identical names; I think @JasonGross 's 4 choices are a good example on why using something different may lead to confusion.

**Review by ejgallego — APPROVED** [review]
Looks pretty good to me, thanks!

My own feeling about this is that this is a somewhat complicated issue; I'm unsure it is possible to codify a set of rules but in the end, given the diversity of projects we may deal with a case-by-case strategy may be more effective.

For example I fully agree that modifications in developments such as f-c-l may be impossible to do without non-trivial help of their developers; I also agree that it is not reasonable for us to become maintainers of a large quantity of 3rd party projects.

On the other hand, I can understand that downstream developers may complain if we break the compatibility so much, and indeed our compat story is severely lacking in some aspects.

Thus, more than for a set of fixed rules I'd advocate trying to be flexible and friendly to downstream projects, and collaborating with them closely in fixing compat issues; I do believe this is already the actual situation indeed.

**Comment by ejgallego** [comment]
Maybe adding some notes about trying to adopt custom compat solutions with project would be worth it?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -527,10 +527,20 @@ how you can anticipate the results of the CI before opening a PR.
...
-should be merged in the broken projects *before* your PR to the Coq
-repository can be.  You can use the same documentation as above to
-learn about testing and fixing locally the broken libraries.
+Furthermore, a backward-compatible fix should be found, properly
+documented in the changelog when non-obvious, and this fix should be
+merged in the broken projects *before* your PR to the Coq repository
+can be.
+
+Note that once the breaking change is well understood, it should not
+feel like it is your role to fix every project that is affected: as
```
Comment (SkySkimmer) [inline CONTRIBUTING.md]: What about before it's well understood?

**Comment by Zimmi48** [comment]
> Maybe adding some notes about trying to adopt custom compat solutions with project would be worth it?

WDYM precisely?

File: CONTRIBUTING.md (hunk 3 above)
Comment (Zimmi48) [reply CONTRIBUTING.md]: Contributors can request help at any time, but we shouldn't go and request maintainers to fix their project if we are not sure what the change here and whether it is intended.

**Comment by ejgallego** [comment]
> WDYM precisely?

IIRC I was thinking about working together in case compat breaks? I dunno, doesn't seem so important.

Not a lot of discussion is happening, I suggest we leave one extra week and merge if no further comments.

**Comment by ejgallego** [comment]
As a post-merge comment, I am getting the feeling that our "guides" are getting a bit long / verbose; I dunno what others think, IMHO we should try to synthesize them a bit, but until our policies / tooling is more stable.

**Comment by Zimmi48** [comment]
At least for the contributing guide, this is on purpose. This is an attempt to provide explicit documentation on the current state of our process. It is important both to the newcomers, for which many things are not obvious, and for main contributors / core developers, because sometimes process get adopted *de facto* but some developers are not aware. I hope that the table of contents compensates for the length by making it easy to quickly find what one is looking for (much like in a reference manual).

**Pull request merged**